File: palindrome/palindrome.js
// Stacks
// first in last out

// using stacks to build a palindrome

let arr = []; //stack

let word = "racecar"
let  rword = "";

for(let i = 0; i<word.length; i++) {
    arr.push(word[i]);
}

console.log(arr);

for(let i = 0; i < word.length; i++) {
    rword+= arr.pop();
}

console.log(rword);

File: palindrome/palindrome.py
class Palindrome:
    def __init__(self, word):
        self.word = word
    def isPalindrome(self):
        wordArray = []
        reverseWord = ""
        i = 0
        for i in  range(len(self.word)):
            wordArray.append(self.word[i])
        j = len(wordArray)
        while j>0:
            reverseWord += wordArray.pop()
            j -= 1
        if(reverseWord == self.word):
            return True
        return False

pal = Palindrome("racecar")
if pal.isPalindrome():
    print("Word is a palindrome")
else:
    print("Word is not a palindrome")